src/Containers/Article/Edit/editArticle.jsx

```diff
@@ -0,0 +1,294 @@
+      formData.append('api_key', process.env.REACT_APP_CLOUDINARY_API_KEY);
+      axios
+        .post(process.env.REACT_APP_CLOUDINARY_URL, formData, {
+          headers: { 'X-Requested-With': 'XMLHttpRequest' },
+        })
+        .then(response => {
+          const { data } = response;
+          spinner = false;
+          images = [...images, data.url];
+          if (data.url) {
+            this.props.setEditorState(
+              addNewBlock(this.props.getEditorState(), Block.IMAGE, {
+                src: data.url,
+              }),
+            );
+          }
+        });
+    }
+    this.props.close();
+  };
+}
+
+/**
+ *
+ * @param {e} e
+ * @returns {method} render
+ */
+export class EditArticle extends Component {
+  state = {
+    title: '',
+    editorState: createEditorState(),
+    loadingBarProgress: 0,
+    contentSetted: false,
+  };
+
+  sideButtons = [
+    {
+      title: 'Image',
+      component: CustomImageSideButton,
+    },
+  ];
+
+  /**
+   * @return {object}
+   * s {jsx} react fragment
+   */
+  componentDidMount() {
+    this.setState({ publish: false });
+    this.refs.editor.focus();
+    const {
+      onGetUserInfo,
+      currentUser,
+      onGetArticle,
+      match: {
+        params: { slug },
+      },
+    } = this.props;
+    onGetUserInfo(currentUser.user.username);
+    onGetArticle(slug);
+  }
+
+  /**
+   * @param {object} props
+   * @return {object}
+   * s {jsx} react fragment
+   */
+  componentWillUpdate = async ({ article, history }) => {
+    const { publish, contentSetted } = this.state;
+    if (article.title) {
+      const { title, body } = article;
+      if (!contentSetted) {
+        await setTimeout(() => {
+          this.setState({
+            loadingBarProgress: 1000,
+            title,
+            editorState: createEditorState(convertToRaw(mediumDraftImporter(JSON.parse(body)))),
+            contentSetted: true,
+          });
+        }, 1000);
+      }
+      if (publish) {
+        history.push(`/articles/${article.slug}`);
+      }
+    }
+  };
+
+  /**
+   *
+   * @param {object} editorState
+   * @returns {method} render
+   */
+  draftOnChange = editorState => {
+    this.setState({ editorState });
+  };
+
+  /**
+   *
+   * @param {object} editorState
+   * @returns {method} render
+   */
+  handleInput = ({ target: { name, value } }) => {
+    this.setState({ [name]: value });
+  };
+
+  getArticleContent = () => {
+    const { title, editorState } = this.state;
+    let body = mediumDraftExporter(editorState.getCurrentContent());
+    body = JSON.stringify(body);
+    const paragraphs = editorState
+      .getCurrentContent()
+      .getPlainText()
+      .split('\n');
+    const firstParagraph = paragraphs.find(paragraph => paragraph);
+    if (images.length === 0) {
+      return {
+        title,
+        description: firstParagraph,
+        body,
+      };
+    }
+    return {
+      title,
+      description: firstParagraph,
+      body,
+    };
```

Comment:
@dusmel This code returns the same thing in both conditions.

Reply:
I fixed that thanks.